internal/remoteconfig/remoteconfig.go

```diff
@@ -350,7 +352,7 @@ func (c *Client) newUpdateRequest() (bytes.Buffer, error) {
 	for i := range c.Capabilities {
 		capa.SetBit(capa, int(i), 1)
 	}
-	products := make([]string, len(c.Products))
+	var products []string
```

Comment:
As far as I can tell, this was good.
Worst case you could use `products := make([]string, 0, len(c.Products))` to preallocate the slice with the expected capacity up front.

Reply:
It wasn't, that's what added some empty products. Your suggestion should be the cleaner fix though, thanks 👍

Reply:
1ad99885c24be24695e